spec.bs

```diff
@@ -8159,43 +8324,151 @@ To <dfn>batch or fetch trusted bidding signals</dfn> given a [=trusted bidding s
+        |policyContainer|.
+      1. If |ig|'s [=interest group/trusted bidding signals keys=] is not null, set
+        |trustedBiddingSignalsBatcher|'s [=trusted bidding signals batcher/keys=] to a
+        [=list/clone=] of |ig|'s [=interest group/trusted bidding signals keys=].
+      1. Otherwise, set |trustedBiddingSignalsBatcher|'s [=trusted bidding signals batcher/keys=] to a
+        new [=list=].
       1. Set |trustedBiddingSignalsBatcher|'s [=trusted bidding signals batcher/ig names=] to
-        |putativeIgNames|.
+          « |igName| ».
       1. Set |trustedBiddingSignalsBatcher|'s [=trusted bidding signals batcher/length limit=] to
-        |putativeLengthLimit|.
+        |ig|'s [=interest group/max trusted bidding signals url length=].
+  1. Otherwise:
+    1. Let |keyValueInterestGroups| be |trustedBiddingSignalsBatcher|'s [=trusted bidding signals
+      batcher/key value interest groups=].
+    1. Let |key| be [=tuple=] of (|ig|'s [=interest group/joining origin=], |ig|'s [=interest group/owner=]).
+    1. If |keyValueInterestGroups| [=map/contains=] |key|, [=set/append=] |ig| to
+      |keyValueInterestGroups|[|key|].
+    1. Otherwise, [=map/set=] |keyValueInterestGroups|[|key|] to a [=set=] « |ig| ».
+
+</div>
+
+<div algorithm>
+
+To <dfn>build trusted key value bidding signals request body</dfn> given a [=set=] |interestGroups|,
+an {{unsigned short}}-or-null |experimentGroupId|, an [=origin=] |topLevelOrigin|, a [=string=]
+|slotSizeQueryParam|, an [=origin=] |coordinator|, and an [=origin=] |owner|:
+
+1. Let |compressionGroups| be an empty [=list=], whose [=list/items=] are [=bidding compression groups=].
+1. Let |compressionGroupMap| be an empty [=map=], where the [=map/keys=] are integers as
+  [=bidding compression group/compression group id=], and the [=map/values=] are [=maps=] with integers as [=map/keys=]
+  for [=bidding partition/id=], and [=bidding partitions=] as their [=map/values=].
+1. Let |compressionIdMap| be an empty [=map=], whose [=map/keys=] are [=origins=] and [=map/values=] are integers.
+1. Let |interestGroupIdMap| be an empty [=map=], whose [=map/keys=] are [=strings=] and [=map/values=] are [=tuples=]
+  of (interger, integer).
+1. Let |slotSizeParams| be the result of [=strictly splitting=] |slotSizeQueryParam| on U+003D (=).
+1. Let |nextCompressionGroupId| be 0.
+1. [=list/For each=] |group| of |interestGroups|:
+  1. Let |joiningOrigin| be |group|'s [=interest group/joining origin=].
+  1. If |compressionIdMap| does not [=map/contain=] |joiningOrigin|:
+    1. [=map/Set=] |compressionIdMap|[|joiningOrigin|] to |nextCompressionGroupId|.
+    1. Increment |nextCompressionGroupId| by 1.
+  1. Let |compressionGroupId| be |compressionIdMap|[|joiningOrigin|].
+  1. If |compressionGroupMap| does not [=map/contain=] |compressionGroupId|:
+    1.[=map/Set=] |compressionGroupMap|[|compressionGroupId|] to an empty [=map=], whose [=map/keys=] are integers

```

Comment:
Need a space, otherwise, it's not rendered as a step correctly
```suggestion
    1. [=map/Set=] |compressionGroupMap|[|compressionGroupId|] to an empty [=map=], whose [=map/keys=] are integers
```

Reply:
Good catch!